### PR TITLE
Fix a few minor typos

### PR DESCRIPTION
**Pull request by tobiasdiez**
Fix a few mior spelling mistakes and add more exceptions to the spellchecker.

### :memo: Checklist

- [ ] The title is concise and informative.
- [ ] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation preview.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -24,10 +24,90 @@
...
+        "codim",
+        "codomain",
+        "coframe",
+        "coframes",
         "Conda",
+        "cputime",
+        "cysignals",
+        "Cython",
+        "d'Alembertian",
+        "dalembertian",
```
Comment (gmou3) [inline .vscode/settings.json:49]:
```suggestion
```

File: .vscode/settings.json [hunk 2]
```diff
@@ -24,10 +24,90 @@
...
+        "doctests",
+        "emptyset",
+        "figsize",
+        "Florentin",
+        "fontsize",
+        "forall",
+        "furo",
+        "Gourgoulhon",
+        "grayskull",
+        "groebner",
```
Comment (gmou3) [inline .vscode/settings.json:61]:
```suggestion
        "Gröbner",
```
like Möbius?

File: src/sage/manifolds/vector_bundle.py [hunk 3]
```diff
@@ -796,7 +796,7 @@ def local_frame(self, *args, **kwargs):
                 resu._init_from_family(sections)
             except ArithmeticError as err:
                 linked = str(err) in ["non-invertible matrix",
-                                      "input matrix must be nonsingular"]
+                                      "input matrix must be non-singular"]
```
Comment (gmou3) [inline src/sage/manifolds/vector_bundle.py]:
```suggestion
                                      "input matrix must be nonsingular"]
```
I prefer nonsingular, and, similarly, nonnegative etc. It is also written without a hyphen on wiki and wolfram.

**Review by gmou3 — CHANGES_REQUESTED** [review]
minor* (since we're at it)

File: .vscode/settings.json (hunk 1 above)
Comment (tobiasdiez) [reply .vscode/settings.json:49]: It's used like this in a few method/variable names, which I think is okay.

File: .vscode/settings.json (hunk 2 above)
Comment (tobiasdiez) [reply .vscode/settings.json:61]: same reason as above

**Review by gmou3 — APPROVED** [review]

**Pull request merged**